src/dymoprint/dymo_print_engines.py

```diff
@@ -119,6 +119,96 @@ def render_barcode(
         )
         return code_bitmap
 
+    def __create_custom_code_bitmap(self, barcode_input_text, bar_code_type):
+        """
+        Renders a barcode image from the input text and barcode type.
+
+        Args:
+            barcode_input_text (str): The input text to be encoded in the barcode.
+            bar_code_type (str): The type of barcode to be rendered.
+
+        Returns:
+            Image: A barcode image.
+        """
+        label_height = DymoLabeler.max_bytes_per_line(self.tape_size_mm) * 8
+        if len(barcode_input_text) == 0:
+            return Image.new("1", (1, label_height))
+
+        code = barcode_module.get(
+            bar_code_type, barcode_input_text, writer=BarcodeImageWriter()
+        )
+
+        code_bitmap = code.render(
+            {
+                "font_size": 10,
+                "vertical_margin": 2,
+                "module_height": (DymoLabeler.max_bytes_per_line(self.tape_size_mm) * 6)
+                - 16,
+                "module_width": 1,
+                "background": "black",
+                "foreground": "white",
+            }
+        )
+        return code_bitmap
+
+    def render_barcode_with_text(
+        self,
+        barcode_input_text,
+        bar_code_type,
+        font_file_name: str,
+        frame_width,
+        font_size_ratio=0.9,
+        align="center",
+    ):
+        """
+        Renders a barcode image with the text below it.
+
+        Args:
+            barcode_input_text (str): The input text to be encoded in the barcode.
+            bar_code_type (str): The type of barcode to be rendered.
+            font_file_name (str): The name of the font file to be used.
+            frame_width (int): The width of the frame around the text.
+            font_size_ratio (float): The ratio of font size to line height. Default
+                is 1.
+
+        Returns:
+            Image: A barcode with text image.
+        """
+        assert align in ("left", "center", "right")
+        code_bitmap = self.__create_custom_code_bitmap(
+            barcode_input_text, bar_code_type
+        )
```

Comment:
This is not necessary, but I am wondering if it would be simpler to just use `render_barcode` here instead of creating the new `__create_custom_code_bitmap` function.

Presumably the barcode wouldn't leave any room for the text in this case, but you could instead make space by drawing a white box where the text is supposed to go. I think that most barcodes are drawn like this anyways.